lib/elixir/lib/task/supervised.ex

```diff
@@ -379,15 +371,22 @@ defmodule Task.Supervised do
         ref = Process.monitor(pid)
         timer_ref = Process.send_after(self(), {:timeout, {monitor_ref, ref}}, timeout)
         send(parent_pid, {:spawned, {monitor_ref, position}, pid})
-        running_tasks = Map.put(running_tasks, ref, {position, type, pid, timer_ref})
+        task_info = %{
+          position: position,
+          type: type,
+          pid: pid,
+          timer_ref: timer_ref,
+          timed_out?: false,
```

Comment:
Mh. I think I prefer the boolean tbh: my reasoning is that when you start a task it feels weird to already say `exit: :down`, given the task is not exited. But it doesn't feel weird to say "this has not timed out" to me. Also, using `:exit` would mean that we would have to use `:down` when starting task and `:timed_out` when a timeout occurs, spreading the atoms we use for these kinds of messages in two places instead of taking the decision on which one only in one place (when task goes down). What do you think? If you still prefer `:exit`, I can of course change the code though :)